pull/github.go

```diff
@@ -103,7 +135,11 @@ func (ghc *GitHubContext) RepositoryName() string {
 }
 
 func (ghc *GitHubContext) Author() (string, error) {
-	return ghc.pr.GetUser().GetLogin(), nil
+	pr, err := ghc.get()
+	if err != nil {
```

Comment:
Should we just load the PR at construction time? Maybe `v3PullRequestToV4` can also take in the Github client necessary to populate the `v4` object if there is insufficient information in the v3 object, instead of returning `nil`.

In that case, we'd probably want to move the `case` checks into a helper:
```
v3PullRequestToV4(...):
    if isSufficient(v3) {
      // seed from v4
    }

    // else graphql query
```

Reply:
I introduced the concept of a `Locator` which contains an optional `github.PullRequest`. If that is missing or incomplete, the full value is now loaded at construction time. I also split out the completeness test as you suggested.